src/main/java/com/michelin/kafkactl/command/Delete.java

```diff
@@ -69,8 +72,9 @@ public Integer onAuthSuccess() {
                 .map(resource -> {
                     ApiResource apiResource =
                         apiResourcesService.getResourceDefinitionByKind(resource.getKind()).orElseThrow();
-                    return resourceService.delete(apiResource, namespace, resource.getMetadata().getName(), dryRun,
-                        commandSpec);
+                    Map<String, Object> spec = resource.getSpec();
+                    return resourceService.delete(apiResource, namespace, resource.getMetadata().getName(),
+                        (spec != null ? (String) spec.get(VERSION) : null), dryRun, commandSpec);
```

Comment:
I think you can directly access `config.nameConfig.version` here. No need to add it in each resource, right?

Reply:
`config.nameConfig.version` instead of `spec.get(VERSION)` you mean? Just above, we use `resource.getMetadata().getName()` to get the name of the resource or `resource.getKind()` so I implemented the version in the same way (ie. building the resource object in `parseResources(namespace)` with the command inputs and accessing it in that object)